### PR TITLE
API auth

### PR DESCRIPTION
**Pull request by dheniges**
- Local login setup with dummy username/pass for now
- JWT auth setup for authenticated endpoints
- Controller loader to apply middleware. Makes adding/removing/updating middleware changeable in a single spot

### CODE REVIEW

File: api/src/controllers/login.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import auth from '../libs/auth'
+import tokenIntegrator from '../integrators/token'
+
+// Caution: login controller routes have NO AUTH by default!
+
+const loginController = function(router) {
+  router.post('/', auth.loginMiddleware, login)
+}
+
+const login = async function(req, res) {
```
Comment (jjidt) [inline api/src/controllers/login.js]: doesn't look like this function needs to be async, or is that planning ahead for a future implementation?

**Review by jjidt — DISMISSED** [review]
looks/works great!
Just the one comment, fix if necessary, but if not merge away 👍

File: api/src/controllers/login.js (hunk 1 above)
Comment (dheniges) [reply api/src/controllers/login.js]: hah, no, not planning ahead, just getting a little async happy

**Comment by dheniges** [comment]
@jjidt still need your official review approval before it's mergeable. Fixed some merge conflicts and removed the async

**Review by jjidt — APPROVED** [review]
👍 nice

**Pull request merged**